Handle macOS 11 and later properly. Based on patch in libtool port.
--- configure.orig	2021-10-27 12:30:23.000000000 -0400
+++ configure	2021-10-27 12:37:51.000000000 -0400
@@ -9211,9 +9211,9 @@
       case ${MACOSX_DEPLOYMENT_TARGET-10.0},$host in
 	10.0,*86*-darwin8*|10.0,*-darwin[91]*)
 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
-	10.[012][,.]*)
+	10.[012][,.]*|,*powerpc*)
 	  _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
-	10.*)
+	*)
 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
       esac
     ;;
